libraries/AP_OpticalFlow/OpticalFlow.cpp

```diff
@@ -154,7 +154,10 @@ void OpticalFlow::update(void)
     if (!enabled()) {
         return;
     }
-
+    if (backend == nullptr && (OpticalFlowType)_type.get() == OpticalFlowType::UAVCAN) {
```

Comment:
We discussed this on the dev call and @tridge suggested that if the _TYPE = UAVCAN then we should always create the driver but it can be unhealthy which should then be caught in the pre-arm checks.